### PR TITLE
Enable eager fetching of maps

### PR DESCRIPTION
**Pull request by joshmoore**
Since the MapField properties are of little use
if the map itself has not been loaded, they are
now actively loaded. Queries need not include a
"left outer join fetch" clause.

Testing:
 * integration tests should be green.
 * some of @dominikl's work should pass **without** gh-3325

### CODE REVIEW

File: components/tools/OmeroPy/test/integration/test_model51.py [hunk 1]
```diff
@@ -100,3 +100,55 @@ def testAsMapMethod(self):
...
+
+    def testMapEagerFetch(self):
+        m = omero.model.MapAnnotationI()
+        m.setMapValue(
+            [NV("foo", "bar")]
+        )
+        m = self.update.saveAndReturnObject(m)
+        anns = self.query.findAllByQuery(
+            "select m from MapAnnotation m ",
+            None)
```
Comment (mtbc) [inline components/tools/OmeroPy/test/integration/test_model51.py]: Would it be a problem here if some other integration testing had left other map annotations around?

**Comment by mtbc** [comment]
Looks like next integration test build indeed ought not have #3325 included.

File: components/tools/OmeroPy/test/integration/test_model51.py (hunk 1 above)
Comment (joshmoore) [inline components/tools/OmeroPy/test/integration/test_model51.py]: This should be a new user in a new group, but I'm happy to add filtering. I wasn't quite sure of the status of @sbesson's re-working to know what the new "Right Way" was (to be).

Comment (mtbc) [inline components/tools/OmeroPy/test/integration/test_model51.py]: Ah, yes, group filtering, good point. I'm satisfied anyway. :smiley:

Comment (sbesson) [inline components/tools/OmeroPy/test/integration/test_model51.py]: See https://github.com/openmicroscopy/openmicroscopy/pull/3318/files#diff-cc72fc55d66d83492366c49e1420fff4R699 for instance. With the new `ITest` class, `self.client` will be shared by all test methods in the class. If the test has any assumption, it is possible to create `new_user_and_client` or `new_client` using the library methods and use this client instead.

**Comment by mtbc** [comment]
Presently we have a test failure in https://ci.openmicroscopy.org/job/OMERO-5.1-merge-integration-python/lastCompletedBuild/testReport/test.integration.test_model51/TestModel51/testMapEagerFetch/

**Comment by joshmoore** [comment]
Pushed.

**Comment by mtbc** [comment]
Well, that seems to have helped the tests at least!

**Comment by will-moore** [comment]
Sorry - forgot to review this yesterday. It's certainly working for me in my current Map Annotation PR and gateway tests. Good to merge from me.

**Comment by joshmoore** [comment]
Merging. If we see any issues with slow client performance, we may have to re-evaluate.

**Comment by joshmoore** [comment]
--no-rebase

**Pull request merged**